### PR TITLE
Model Hosting Revamp

### PR DESCRIPTION
**Pull request by rasswanth-s**
## Description
Please include a summary of the change, the motivation, and any additional context that will help others understand your PR. If it closes one or more open issues, [please tag them as described here](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

File: notebooks/experimental/enclaves/V3/V3-Enclave-Model-HostingSingle-Notebook.ipynb
Comment (yashgorana) [inline notebooks/experimental/enclaves/V3/V3-Enclave-Model-HostingSingle-Notebook.ipynb:1]: Can we remove the "*.safetensors" from ignore_patterns? That way the whole notebook will be e2e runnable. currently this isn't running in CI so runtime shouldn't be an issue. but we should investigate why the asset are taking a lot of time to get ready as we discussed on slack.

File: packages/syft/src/syft/service/action/action_service.py
Comment (yashgorana) [inline packages/syft/src/syft/service/action/action_service.py:1]: Please run action object tests once locally before we merge this change - don't want stuff to break as this PR doesn't run the CI :)

File: notebooks/experimental/enclaves/V3/V3-Enclave-Model-HostingSingle-Notebook.ipynb
Comment (rasswanth-s) [reply notebooks/experimental/enclaves/V3/V3-Enclave-Model-HostingSingle-Notebook.ipynb:1]: Done 👍

File: packages/syft/src/syft/service/action/action_service.py
Comment (rasswanth-s) [reply packages/syft/src/syft/service/action/action_service.py:1]: Ran all the action object tests by tox -e syft.test.unit.
All the action objects passed ✅ 

There are a few tests ,which are failing for single domain user code, which are not related to this current PR , but due to the main PR.

**Pull request merged**